teps/0057-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,514 @@
+- Create a standard around how to refer to Tasks in the catalog from Pipelines in the catalog
+- Create a standard around how to refer to Tasks which may not may be part of catalog but are used by Pipelines in the catalog
+
+### Non-Goals
+
+- Infrastructure related to the catalog (test infrastructure, …)
+
+- Tools related to catalog maintenance (tekdoc, linter, …)
+
+- Tool to install pipeline and all associated resources
+
+- A tool to display deprecation warnings if a dependent task is deprecated.
+
+### Use Cases (optional)
+
+- Having a pipeline which builds, pushes and deploys an application for a particular language.
+
+- As a CI which can be used to run the test cases, lint checks
+
+- Packaging related CD activities for a particular language, e.g. for publishing a go binary could include testing, linting, building
+
+## Requirements
+
+### Quality
+
+1. Resources should be well documented:
+
+   1. All configuration options are documented
+
+   2. Examples are provided
+
+2. If a resource is in the catalog, a user should feel confident that
+   it will work as advertised See [Tekton Catalog Test Infrastructure for
+   this](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)
+
+3. When pipelines have parameters and/or results, it must be
+   clear what values are allowed and the user must receive feedback
+   when they provide values that are not valid
+
+## Proposal
+
+### Support Tiers
+
+This proposal includes [three support
+levels](https://github.com/tektoncd/catalog/issues/5):
+
+1. **Community**
+2. **Verified**
+3. **Official**
+
+Those support tiers are mainly aimed for the Tekton Hub.
+
+For more detailed information about Support Tiers please see [here](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#support-tiers)
+
+### Versioning Resources
+
+Resources definition evolves across time. A new version of the image used is published,
+a new feature in tektoncd/pipeline is available, or a new feature / behaviour is added.
+On the other hand, the catalog should guarantee as much stability as possible for its user.
+
+A version is an identifiable information of a resource along with its kind (Task, Pipeline)
+and name, and it should be present in the definition of the resource. The location of a resource
+in the catalog should be computable given its Kind, Name and the Version.
+e.g. `/{kind}/{name}/{version}/{name}.yaml` - `pipeline/<pipeline-name>/0.5/<pipeline-name>.yaml`
+
+Adding a new version of a resource should be considered as a release of that resource. Here are some cases where a version bump for pipeline may be required
+
+- Task used in the pipeline has been deprecated/removed
+- Location of the task used in the pipeline is changed
+- If there's a new version of the task then it completely depends on the author
+  of the pipeline whether to bump the version of the pipeline or not
```

Comment:
i want to ask a bit more about this - to avoid blocking this proposal, maybe we could remove this (or say it is TDB) and have a separate TEP dealing with when to bump the version of items in the catalog in general?

im a bit confused by the idea of changing something in a pipeline or task without bumping the version number - im wondering if the motivation for this might be that we made it too hard to bump the version in the catalog? because i think the clearest and least confusing thing to do is to bump the version whenever we make a change, which is the same approach we would take to any other software (e.g. if we did a release of tekton pipelines where we'd changed the version of a dependency, even if it was to bring in a critical bug fix, we'd bump the version, we wouldn't re-use the same version number)

i.e. folks should know that they can pin to a certain version number without having things change out from under them

( + @jerop )

Reply:
could we add this maybe to the questions section below for now?